### PR TITLE
add doc folder for vuepress doc

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do ?

:warning: This PR must be merge `before` [#403](https://github.com/kuzzleio/documentation/pull/403) :warning: 

Create a `/doc` folder with symlinked readme.md and /img folders to allow vuepress doc display the how to.

Add doc.sh script 

In this PR you can read the How To's but you can't access to the page with tiles.

-> npm run doc-prepare
-> npm run doc-dev
-> localhost:8080/how-to/1/kuzzle-esp32 (or other pages)

### CODE REVIEW

File: doc/doc.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    echo "Install dependencies"
+    npm --prefix framework/ install
+  ;;
+
+  dev)
+    ./framework/node_modules/.bin/vuepress dev $DOC_VERSION/ $ARGS
+  ;;
+
+  build)
```
Comment (Aschen) [inline doc/doc.sh]: Missing the `build-netlify` script

File: doc/doc.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    then
+      echo "Update documentation framework"
+      git -C framework/ pull origin master
+    else
+      echo "Clone documentation framework"
+      git clone --depth 10 --single-branch --branch master https://github.com/kuzzleio/documentation.git framework/
+    fi
+
+    echo "Link local doc for dead links checking"
+    rm framework/src$DOC_PATH
```
Comment (Aschen) [inline doc/doc.sh]:
```suggestion
    rm -f framework/src$DOC_PATH
```

File: doc/doc.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+if [ ! -d "./$DOC_DIR" ]
+then
+  echo "Cannot find $DOC_DIR/. You must run this script from doc/ directory."
+  exit 1
+fi
+
+case $1 in
+  prepare)
+    if [ -d "framework" ];
```
Comment (Aschen) [inline doc/doc.sh]: We didn't try to updat the repo anymore, instead we do a `rm -rf` of the old framework

File: package.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "scripts": {
+        "doc-prepare": "cd doc && bash doc.sh prepare",
+        "doc-dev": "cd doc && bash doc.sh dev",
+        "doc-build": "cd doc && bash doc.sh build",
+        "doc-upload": "cd doc && bash doc.sh upload",
+        "doc-cloudfront": "cd doc && bash doc.sh cloudfront",
+        "doc-deploy": "npm run doc-upload && npm run doc-cloudfront"
```
Comment (Aschen) [inline package.json]: Missing `doc-netlify` script

File: doc/doc.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+if [ ! -d "./$DOC_DIR" ]
+then
+  echo "Cannot find $DOC_DIR/. You must run this script from doc/ directory."
+  exit 1
+fi
+
+case $1 in
+  prepare)
+    echo "Clone documentation framework"
+    git clone --depth 10 --single-branch --branch master https://github.com/kuzzleio/documentation.git framework/
```
Comment (Aschen) [inline doc/doc.sh]:
```suggestion
    rm -rf framework/
    git clone --depth 10 --single-branch --branch master https://github.com/kuzzleio/documentation.git framework/
```

File: doc/doc.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    ./framework/node_modules/.bin/vuepress build $DOC_VERSION/ $ARGS
+  ;;
+
+
+  upload)
+    aws s3 sync $DOC_VERSION/.vuepress/dist s3://$S3_BUCKET$SITE_BASE --delete
+  ;;
+
+  cloudfront)
+    aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths $SITE_BASE
```
Comment (Aschen) [inline doc/doc.sh]:
```suggestion
    aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths $SITE_BASE*
```
Missing wildcard to matche every file besides `/how-to/1/*`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+    "scripts": {
+        "doc-prepare": "cd doc && bash doc.sh prepare",
+        "doc-dev": "cd doc && bash doc.sh dev",
+        "doc-build": "cd doc && bash doc.sh build",
+        "doc-build-netlify": "cd doc && bash doc.sh build-netlify",
```
Comment (Aschen) [inline package.json]:
```suggestion
        "doc-netlify": "cd doc && bash doc.sh build-netlify",
```
Keep the same name in all package.json

File: keep-only-warm-data/README.md [hunk 8]
```diff
@@ -1,3 +1,10 @@
+---
+code: true
+type: page
+title: Keep only warm data
+description: How to keeping only warm data in Kuzzle
```
Comment (scottinet) [inline keep-only-warm-data/README.md]:
```suggestion
description: How to keep only warm data in Kuzzle
```

File: sync-data-to-another-database/README.md [hunk 9]
```diff
@@ -1,3 +1,11 @@
+---
+code: true
+type: page
+title: Sync to another database
+description: How to synchronise data to another database
```
Comment (scottinet) [inline sync-data-to-another-database/README.md]:
```suggestion
description: How to synchronize data to another database
```

**Pull request merged**